src/Subscriber/FindReviewerSubscriber.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace App\Subscriber;
+
+use App\Api\PullRequest\PullRequestApi;
+use App\Command\SuggestReviewerCommand;
+use App\Entity\Task;
+use App\Event\GitHubEvent;
+use App\GitHubEvents;
+use App\Service\TaskScheduler;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class FindReviewerSubscriber implements EventSubscriberInterface
+{
+    private $pullRequestApi;
+    private $botUsername;
+    private $scheduler;
+
+    public function __construct(PullRequestApi $pullRequestApi, TaskScheduler $scheduler, string $botUsername)
+    {
+        $this->pullRequestApi = $pullRequestApi;
+        $this->botUsername = $botUsername;
+        $this->scheduler = $scheduler;
+    }
+
+    public function onPullRequest(GitHubEvent $event)
+    {
+        $data = $event->getData();
+        if (!in_array($data['action'], ['opened', 'ready_for_review']) || ($data['pull_request']['draft'] ?? false)) {
+            return;
+        }
+
+        $repository = $event->getRepository();
+
+        // set scheduled task to run in 20 hours
+        $this->scheduler->runLater($repository, $data['number'], Task::ACTION_SUGGEST_REVIEWER, new \DateTimeImmutable('+20hours'));
+    }
+
+    /**
+     * When somebody make a comment on an issue or pull request.
+     * If it includes the bot name and "review" on the same line, then try to find reviewer.
+     */
+    public function onComment(GitHubEvent $event)
+    {
+        $data = $event->getData();
+        if ('created' !== $data['action']) {
+            return;
+        }
+
+        if (false === strpos($data['comment']['body'], $this->botUsername)) {
+            return;
+        }
+
+        // Search for "review"
+        $lines = explode(PHP_EOL, $data['comment']['body']);
+        foreach ($lines as $line) {
+            if (false === strpos($line, $this->botUsername)) {
+                continue;
+            }
+
+            if (false === strpos($line, 'review')) {
+                continue;
+            }
+
+            $number = $data['issue']['number'];
+            $this->pullRequestApi->findReviewer($event->getRepository(), $number, SuggestReviewerCommand::TYPE_DEMAND);
+            $event->setResponseData([
+                'issue' => $number,
+                'suggest-review' => true,
+            ]);
+
+            return;
+        }
```

Comment:
FYI @javiereguiluz: Here is the "text command" parsing.

Reply:
Very nice! It follows the *"be liberal in what you accept"* principle.

Reply:
Should we do `preg_match('/\breview\b/', ...)` to prevent `@carsonbot reviewed` from being matched? (e.g. in case someone doesn't understand how to change the status label)

Reply:
Done. Im using `preg_match` instead. I added a bunch more tests. The same rules apply. If one is using `@carsonbot` and (`review` or `reviewer`) on the same line, then we will trigger a search for a candidate.